Products/ZenModel/ZenPack.py

```diff
@@ -1382,37 +1383,53 @@ def installServiceDefinitions(self, serviceDefs, servicePaths):
         if isinstance(servicePaths, basestring):
             servicePaths = [servicePaths]
 
+        serviceDefs = [json.loads(sd) for sd in serviceDefs]
+        for sd, sp in zip(serviceDefs, servicePaths):
+            sd["parentServicePath"] = sp
+
+        for sd in serviceDefs:
+            sd["Services"] = []
+            sp = sd["parentServicePath"]
+            sp = "" if sp == "/" else sp
+            paths = [sp + "/=" + sd["Name"]]
+            for tag in sd["Tags"]:
+                paths.append(sp + "/" + tag)
+            sd["servicePaths"] = paths
```

Comment:
What if, instead of adding 'servicePaths' to the service dictionary, you have a separate map: servicePath -> service.  That's really what you want.  It would make the look up code cleaner and you wouldn't have to remember to remove the entry from the service.

You could make a similar argument about the parentServicePath.  Should that be a map: service -> parentPath?